Directory.Build.targets

```diff
@@ -34,4 +41,29 @@
     <PackageReference Include="Microsoft.DotNet.XUnitExtensions" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(IsSourceProject)' == 'true'">
+    <AssemblyAttribute Include="System.CLSCompliantAttribute" Condition="'$(CLSCompliant)' == 'true'">
+      <_Parameter1>true</_Parameter1>
+    </AssemblyAttribute>
+    <AssemblyAttribute Include="System.Reflection.AssemblyDefaultAliasAttribute">
+      <_Parameter1>$(AssemblyName)</_Parameter1>
+    </AssemblyAttribute>
+    <AssemblyAttribute Include="System.Reflection.AssemblyDescriptionAttribute">
+      <_Parameter1>$(AssemblyName)</_Parameter1>
+    </AssemblyAttribute>
+    <AssemblyAttribute Condition="'$(MSBuildProjectExtension)' == '.csproj'" Include="System.Runtime.InteropServices.DefaultDllImportSearchPathsAttribute">
+      <_Parameter1>System.Runtime.InteropServices.DllImportSearchPath.AssemblyDirectory | System.Runtime.InteropServices.DllImportSearchPath.System32</_Parameter1>    
+      <_Parameter1_IsLiteral>true</_Parameter1_IsLiteral>    
+    </AssemblyAttribute>  
+    <AssemblyMetadata Include="PreferInbox">
+      <Value>True</Value>
+    </AssemblyMetadata>
+    <AssemblyMetadata Include="Serviceable">
+      <Value>True</Value>
+    </AssemblyMetadata>
+    <AssemblyMetadata Include=".NETFrameworkAssembly">
```

Comment:
As a follow up to this PR can you check if this is still needed with @MichalStrehovsky.  Looks like he removed it here https://github.com/dotnet/runtime/pull/89490.  We might want to find out if we should remove from these old assemblies in servicing as well.  I don't think we need to block this PR on it though.

Reply:
Good catch. While we wait for @MichalStrehovsky 's confirmation, I think the safer thing to do is to imitate runtime and _not_ add this attribute. I'll remove it.

Reply:
What point in time do we capture here? If it's point in time where we used to ship UapAot and e.g. we can service UapAot version of System.Reflection.DispatchProxy from this repo, this attribute should be set. If this captures a point in time after which we nuked all UapAot support from corefx, the attribute is unnecessary.

Reply:
The latter (no UAP). So I will definitely remove it. Thanks.